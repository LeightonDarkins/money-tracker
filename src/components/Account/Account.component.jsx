import React from 'react'
import PropTypes from 'prop-types'
import currency from '../../common/currency'

import './Account.scss'

const Account = ({ openingBalance, balance, name, onAccountClick }) => {
  return (
    <li className='account-list-item' onClick={onAccountClick}>
      <div className='account-container'>
        <span className='account-element account-name'>{ name }</span>
        <span className='account-element account-balance'>{ currency.numberAsCurrency(balance) }</span>
        <div className='account-element account-opening-balance'>Opening Balance: { currency.numberAsCurrency(openingBalance) }</div>
      </div>
    </li>
  )
}

Account.propTypes = {
  balance: PropTypes.number.isRequired,
  openingBalance: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  onAccountClick: PropTypes.func.isRequired
}

export default Account
