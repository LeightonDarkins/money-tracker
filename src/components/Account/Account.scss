@import '../../variables.scss';

.account-container {
  width: 100%;
  height: $default-element-height;
  min-height: $default-account-element-height;
  margin: $default-spacing auto $default-spacing auto;
  border-bottom: $divider-border;
  display: grid;
  grid-template-columns: 75% 25%;
  grid-template-rows: 50% auto;

  .account-element {
    padding: $default-spacing;
  }

  .account-name {
    grid-column-start: 1;
    grid-column-end: 2;
    font-size: $default-font-size;
    vertical-align: middle;
  }

  .account-balance {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    grid-column-start: 2;
    grid-row-start: 1;
    grid-row-end: span 2;
  }

  .account-opening-balance {
    display: flex;
    align-items: flex-end;
    font-size: 13px;
    color: #aaa;
    grid-row-start: 2;
    grid-column-start: 1;
    grid-column-end: span 2;
  }
}
