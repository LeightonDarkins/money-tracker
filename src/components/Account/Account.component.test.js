/* global describe, it, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import Account from './Account.component'

describe('Account', () => {
  describe('Snapshots', () => {
    it('renders correctly', () => {
      const account = renderer.create(<Account
        name='Test Account Name'
        balance={500}
        openingBalance={1000}
        onAccountClick={(y) => y} />).toJSON()
      expect(account).toMatchSnapshot()
    })
  })
})
