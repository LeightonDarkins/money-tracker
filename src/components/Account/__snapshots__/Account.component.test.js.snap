exports[`Account Snapshots renders correctly 1`] = `
<li
  className="account-list-item"
  onClick={[Function]}>
  <div
    className="account-container">
    <span
      className="account-element account-name">
      Test Account Name
    </span>
    <span
      className="account-element account-balance">
      $5.00
    </span>
    <div
      className="account-element account-opening-balance">
      Opening Balance: 
      $10.00
    </div>
  </div>
</li>
`;
